include/openssl/params.h

```diff
@@ -67,6 +67,8 @@ extern "C" {
 # define OSSL_PARAM_octet_ptr(key, addr, sz) \
     OSSL_PARAM_DEFN((key), OSSL_PARAM_OCTET_PTR, (addr), sz)
 
+# define OSSL_PARAM_is_empty(param) (params == NULL || params->key == NULL)
```

Comment:
+1 to that - even if not public, best practice for macro arguments is to put them in parentheses, even if not strictly necessary

Reply:
This macro should really be a function, since it uses `params` twice. Think of certain side effects.

I could see this be a static inline function, but that's not necessary.

Reply:
I'd propose to make it a non-public macro then. Adding another library call into every set_ctx_params() call does not look to me as a good idea.

Reply:
Agreed with a non-public definition.  I'm not fussed if it is a macro or a function.

Reply:
I don't see any harm in making this a function.  LTO will inline it if the compiler thinks it's a good idea and procedure calls are essentially free on modern processors.

Reply:
> I don't see any harm in making this a function. LTO will inline it if the compiler thinks it's a good idea and procedure calls are essentially free on modern processors.

LTO is not used by default.